utils.go

```diff
@@ -15,3 +21,73 @@ func Int64(v int64) *int64 { return &v }
 // String is a helper routine that allocates a new string value
 // to store v and returns a pointer to it.
 func String(v string) *string { return &v }
+
+// elem checks that the first parameter is an element of the received
+// slice.
+func elem(str string, elems []string) bool {
```

Comment:
Just a nitpick, but I think you don't need the `elem` function if you return the boolean fields as a map from string to bool. It would also be slightly faster.

Reply:
Thank you, that's indeed a better approach. I've pushed a commit with the changes.